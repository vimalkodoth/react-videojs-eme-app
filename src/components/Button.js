import React, { Component } from "react";
import { css } from "@emotion/core";
import arrow from "./../static/img/arrow.svg";
import { bool, func } from "prop-types";

/**
 * Button Component
 */
class Button extends Component {
    static propTypes = {
        onClick: func,
        show: bool,
        prev: bool
    };
    static defaultProps = {
        onClick: (val) => val,
        show: true,
        prev: true
    };
    render() {
        const { onClick, show, prev } = this.props;
        return (
            <div
                data-testid="button"
                css={ButtonStyles({ prev })}
                style={{
                    display: show ? "block" : "none"
                }}
                onClick={onClick}
            ></div>
        );
    }
}

export default Button;

const ButtonStyles = (props) => css`
    position: absolute;
    cursor: pointer;
    top: 50%;
    margin-top: -16px;
    width: 32px;
    height: 32px;
    @media (min-width: 560px) {
        margin-top: -24px;
        width: 48px;
        height: 48px;
    }
    background-image: url(${arrow});
    background-color: rgba(255, 255, 0, 0.75);
    background-size: 110%;
    background-repeat: no-repeat;
    background-position: -3px -2px;
    border-radius: 50%;
    z-index: 9999;
    ${props && props.prev
        ? `
    left: 5px;
  `
        : `
    right: 5px;
    transform: rotate(180deg);
  `}
`;
