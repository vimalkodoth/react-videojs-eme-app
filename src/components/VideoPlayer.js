import React from "react";
import { css } from "@emotion/core";
import player from "./../utils/player";

class VideoPlayer extends React.Component {
    player = null;

    componentDidMount() {
        this.player = player.init(
            this.videoNode,
            {
                html5: {
                    nativeCaptions: false
                },
                ...this.props
            },
            () => {
                this.player.play();
            }
        );
    }

    componentWillUnmount() {
        if (this.player) {
            this.player.destroy();
            this.player = null;
        }
    }

    render() {
        return (
            <div>
                <div data-vjs-player>
                    <video
                        ref={(node) => (this.videoNode = node)}
                        className="video-js vjs-theme-rakuten full"
                        css={Player}
                    ></video>
                </div>
            </div>
        );
    }
}

export default VideoPlayer;

const Player = css`
    &.vjs-theme-rakuten {
        max-width: 100%;
        &.full {
            position: absolute;
            left: 0;
            right: 0;
            min-width: 100%;
            height: 100%;
        }
        max-width: 100%;
        min-width: 100%;
        .vjs-big-play-button {
            position: absolute;
            top: 50%;
            left: 50%;
            margin-top: -0.81666em;
            margin-left: -1.5em;
        }
    }
`;
