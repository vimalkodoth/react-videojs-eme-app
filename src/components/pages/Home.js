import React, { Component } from "react";
import MoviesList from "../MoviesList.js";
class Home extends Component {
    render() {
        return (
            <div className="home">
                <MoviesList />
            </div>
        );
    }
}

export default Home;
