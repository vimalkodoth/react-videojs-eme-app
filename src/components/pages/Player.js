import React, {useRef, useEffect, useState, StrictMode} from "react";
import { useDispatch, useSelector } from 'react-redux';
import { css } from "@emotion/core";
import YouTube from 'react-youtube';
import SubmitButton from "../SubmitButton";
import usePlaybackConfiguration from "../hooks/usePlaybackConfiguration";
import useStorePositionTime from "../hooks/useStorePositionTime";

const defaultConfigs = {
    height: '390',
    width: '640',
    playerVars: {
      // https://developers.google.com/youtube/player_parameters
      autoplay: 1,
      start: 0
    }
};

function Player(props) {
    const playerRef = useRef(null);
    const opts = usePlaybackConfiguration(defaultConfigs);
    const [onPlayerStateChange] = useStorePositionTime(playerRef);
    const history = useHistory();
    const onEdit = () => {
        history.goBack();
    }
    if(!opts) return <div></div>;
    console.log('render');
    return (
        <div css={Wrapper}>
            <YouTube videoId="RrTlbxFhK_I" opts={opts} ref={playerRef} onStateChange={onPlayerStateChange}/>
            <SubmitButton onClick={onEdit}>
            Edit
            </SubmitButton>
            <SubmitButton to={{
                    pathname: `/gif`,
                    state: {}
                }}>
            Next (GIF)
            </SubmitButton>
        </div>
    );
}

export default Player;

const Wrapper = css`
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
    align-items: center;
    border: 1px solid #fff;
`;