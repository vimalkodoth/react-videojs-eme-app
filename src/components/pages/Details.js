import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchMovieDetail } from "./../../actionCreators";
import DetailsHeader from "./../DetailsHeader";
import Button from "./../../components/Button";
import withBackButton from "./../hocs/withBackButton";
import { func, object } from "prop-types";

class Details extends Component {
    static propTypes = {
        onBackButtonClicked: func,
        fetchMovieDetail: func,
        movieDetails: object,
        match: object
    };
    static defaultParams = {
        onBackButtonClicked: (n) => n,
        fetchMovieDetail: (n) => n,
        match: {},
        movieDetails: {}
    };
    componentDidMount() {
        const {
            match: { params }
        } = this.props;
        this.props.fetchMovieDetail(params.id);
    }

    goBack = (e) => {
        this.props.onBackButtonClicked(e);
    };

    render() {
        const { match } = this.props;
        return (
            <div className="details">
                <Button onClick={this.goBack} show={true} prev={true} />
                <DetailsHeader id={match.params.id} />
            </div>
        );
    }
}

const mapStateToProps = (state) => ({ movieDetails: state.movieDetails });
const mapDispatchToProps = (dispatch) => ({
    fetchMovieDetail(id) {
        dispatch(fetchMovieDetail(id));
    }
});

export const UnWrappedDetails = Details;

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withBackButton(Details));
