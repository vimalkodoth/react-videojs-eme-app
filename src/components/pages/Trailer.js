import React, { Component } from "react";
import VideoPlayer from "./../VideoPlayer";
import Button from "./../Button";
import withBackButton from "./../hocs/withBackButton";
import { func } from "prop-types";

//Config needs to be moved from component
const videoJsOptions = {
    src:
        "https://bitmovin-a.akamaihd.net/content/art-of-motion_drm/mpds/11331.mpd",
    autoplay: true,
    controls: true,
    type: "application/dash+xml",
    keySystems: {
        "com.widevine.alpha": "https://widevine-proxy.appspot.com/proxy"
    }
};

/** Trailer Page Component */
class Trailer extends Component {
    static propTypes = {
        onBackButtonClicked: func
    };
    static defaultProps = {
        onBackButtonClicked: (n) => n
    };
    goBack = (e) => {
        this.props.onBackButtonClicked(e);
    };
    render() {
        return (
            <div className="trailer">
                <Button onClick={this.goBack} show={true} prev={true} />
                <VideoPlayer {...videoJsOptions} />;
            </div>
        );
    }
}

export const UnWrappedTrailer = Trailer;

export default withBackButton(Trailer);
