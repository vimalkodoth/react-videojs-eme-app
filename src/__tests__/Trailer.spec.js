import React from "react";
import { shallow } from "enzyme";
import { UnWrappedTrailer } from "./../components/pages/Trailer";
import Button from "./../components/Button";
import VideoPlayer from "./../components/VideoPlayer";

describe("Trailer", () => {
    it("renders a videoplayer and button correctly", () => {
        const component = shallow(<UnWrappedTrailer />);
        expect(component).toMatchSnapshot();
        expect(component.find(VideoPlayer).length).toEqual(1);
        expect(component.find(Button).length).toEqual(1);
    });
});
