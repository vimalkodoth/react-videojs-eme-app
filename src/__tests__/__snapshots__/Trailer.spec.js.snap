// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trailer renders a videoplayer and button correctly 1`] = `ShallowWrapper {}`;
